cmd/fs/helpers.go

```diff
@@ -46,6 +50,75 @@ func filerForPath(ctx context.Context, fullPath string) (filer.Filer, string, er
 	return f, path, err
 }
 
+const dbfsPrefix string = "dbfs:/"
+const volumesPefix string = "dbfs:/Volumes"
+
 func isDbfsPath(path string) bool {
-	return strings.HasPrefix(path, "dbfs:/")
+	return strings.HasPrefix(path, dbfsPrefix)
+}
+
+type validArgs struct {
+	mustWorkspaceClientFunc func(cmd *cobra.Command, args []string) error
+	filerForPathFunc        func(ctx context.Context, fullPath string) (filer.Filer, string, error)
+	pathArgCount            int
+	onlyDirs                bool
+}
+
+func newValidArgs() *validArgs {
+	return &validArgs{
+		mustWorkspaceClientFunc: root.MustWorkspaceClient,
+		filerForPathFunc:        filerForPath,
+		pathArgCount:            1,
+		onlyDirs:                false,
+	}
+}
+
+func (v *validArgs) Validate(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+	cmd.SetContext(root.SkipPrompt(cmd.Context()))
+
+	err := v.mustWorkspaceClientFunc(cmd, args)
+
+	if err != nil {
+		return nil, cobra.ShellCompDirectiveError
+	}
+
+	filer, toCompletePath, err := v.filerForPathFunc(cmd.Context(), toComplete)
+	if err != nil {
+		return nil, cobra.ShellCompDirectiveError
+	}
+
+	completer := completer.New(cmd.Context(), filer, v.onlyDirs)
+
+	if len(args) >= v.pathArgCount {
+		return nil, cobra.ShellCompDirectiveNoFileComp
+	}
+
+	completions, dirPath, directive, err := completer.CompletePath(toCompletePath)
+
+	if err != nil {
+		return nil, cobra.ShellCompDirectiveError
+	}
+
+	isDbfsPath := isDbfsPath(toComplete)
+
+	for i := range completions {
+		completions[i] = filepath.Join(dirPath, completions[i])
+
+		if isDbfsPath {
+			// Add dbfs:/ prefix to completions
+			completions[i] = path.Join(dbfsPrefix, completions[i])
+		}
+	}
+
+	// If the path is a dbfs path, we try to add the dbfs:/Volumes prefix suggestion
+	if isDbfsPath && strings.HasPrefix(volumesPefix, toComplete) {
```

Comment:
Anecdotally, I found that `/Volumes/` and a bunch more aliases are already added by the DBFS list API.

Reply:
Checked and you're right - this can be removed